web/src/util.ts

```diff
@@ -4,8 +4,26 @@ import * as blueslip from "./blueslip";
 import {$t} from "./i18n";
 import type {MatchedMessage, Message, RawMessage, UpdateMessageEvent} from "./types";
 
+// When the type is "private", properties from to_user_ids might be undefined.
+// See https://github.com/zulip/zulip/pull/23032#discussion_r1038480596.
+// Recipient type in direct messages (DM)
+export type DirectRecipient = {
+    type: "private";
+    to_user_ids: string;
```

Comment:
Can we make this field here optional instead of using `PartialBy` later when defining `Recipient` type?

Reply:
We can. But not sure which one is better. `DirectRecipient` and `StreamRecipient` are re-usable in other files like `compose_fade.js` too.